test/unit/clients/connector_client/connector_patch_apple_pay_toggle_test.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+// NPM dependencies
+const Pact = require('pact')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+// Custom dependencies
+const path = require('path')
+const PactInteractionBuilder = require('../../../fixtures/pact_interaction_builder').PactInteractionBuilder
+const Connector = require('../../../../app/services/clients/connector_client').ConnectorClient
+const gatewayAccountFixtures = require('../../../fixtures/gateway_account_fixtures')
+
+// Constants
+const ACCOUNTS_RESOURCE = '/v1/api/accounts'
+const port = Math.floor(Math.random() * 48127) + 1024
+const connectorClient = new Connector(`http://localhost:${port}`)
+const expect = chai.expect
+const existingGatewayAccountId = 666
+
+// Global setup
+chai.use(chaiAsPromised)
+
+describe('connector client - patch apple pay toggle (enabled) request', () => {
+  const request = gatewayAccountFixtures.validGatewayAccountApplePayToggleRequest(true).getPlain()
+  let provider = Pact({
+    consumer: 'selfservice',
+    provider: 'connector',
+    port: port,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  after(() => provider.finalize())
+
+  describe('apple pay toggle - supported payment provider request', () => {
+    const applePayToggleSupportedPaymentProvider =
+      `a gateway account supporting digital wallet with external id ${existingGatewayAccountId} exists in the database`
+
+    before(() => {
+      return provider.addInteraction(
+        new PactInteractionBuilder(`${ACCOUNTS_RESOURCE}/${existingGatewayAccountId}`)
+          .withUponReceiving('a valid patch apple pay toggle (enabled) request')
+          .withState(applePayToggleSupportedPaymentProvider)
+          .withMethod('PATCH')
+          .withRequestBody(request)
+          .withStatusCode(200)
+          .withResponseHeaders({})
+          .build())
+    })
+
+    afterEach(() => provider.verify())
+
+    it('should toggle successfully', done => {
+      connectorClient.toggleApplePay(existingGatewayAccountId, true, null)
+        .should.be.fulfilled
+        .notify(done)
+    })
+  })
+
+  describe('apple pay toggle with unsupported payment provider request', () => {
+    const applePayToggleUnsupportedPaymentProviderState = `User ${existingGatewayAccountId} exists in the database`
+    const unsupportedPaymentProvider = 'sandbox'
+    const errorResponse = {
+      message: `Gateway ${unsupportedPaymentProvider} does not support digital wallets.`
```

Comment:
@stephencdaly raised a good point about not asserting on things in pact tests which we don't actually rely upon. So if we're not using `message` then we should not assert on it. Maybe speak with Stephen as I think he's been removing these fields in pact tests.

Reply:
Awesome! 👍